conjur/variable/variable_logic.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+
+"""
+VariableLogic module
+
+This module is the business logic for executing the VARIABLE command
+"""
+
+# Builtins
+import json
+import logging
+
+# pylint: disable=too-few-public-methods
+class VariableLogic:
+    """
+    VariableLogic
+
+    This class holds the business logic for executing and manipulating
+    returned data
+    """
+    def __init__(self, client):
+        self.client = client
+
+    # pylint: disable=logging-fstring-interpolation
+    def run_action(self, variable_data):
+        """
+        Method to determine which subcommand action to run {get, set}
+        """
+        if variable_data.action == 'get':
+            # pylint: disable=no-else-return
+            logging.debug(f"Fetching variable value(s) for: {variable_data.variable_id}")
+            if len(variable_data.variable_id) == 1:
+                variable_value = self.client.get(variable_data.variable_id[0])
+                return variable_value.decode('utf-8')
+            else:
+                variable_values = self.client.get_many(*variable_data.variable_id)
+                return json.dumps(variable_values, indent=4)
+        else:
```

Comment:
don't think you want the else here. this could lead to unknown behaviour like the user input is "getr" we still treat it the same as "set".
this should be treated more like a switch case pattern (but python so many if's)
![bump_on_the_head](https://user-images.githubusercontent.com/76062747/105174984-02499f00-5b2c-11eb-9bb8-0e21c5b41584.gif)

Reply:
this else is for just validating the length to know if it is a single variable or multiple variables to make a batch request.

Reply:
I was taking about the outer "else". maybe I'm missing but you have `f variable_data.action == 'get':`
and the `else:` referring to it assume this is a set request right? sound pretty dengerous incase of a typo

Reply:
it wont get past argparser because it validates if the action is get or set otherwise it doesnt recognize the commands and errors. That being said, you are correct and i added `elif args.action == 'set':` to be explicit

Reply:
cool.
I think every layer be independent and protect from malicious input.